.github/workflows/gradle.yml

```diff
@@ -91,41 +70,97 @@ jobs:
-      timeout-minutes: 60
-      env:
-        GRADLE_OPTS: "-Dorg.gradle.jvmargs='-XX:MaxMetaspaceSize=512m'"
 # Copy build reports and upload artifact if build failed
-    - name: Copy build/report/tests/test for upload (Linux)
-      if: ${{ runner.os == 'Linux' && failure() }}
+    - name: Copy build/report/tests/test for upload
+      if: ${{ failure() }}
       run: mkdir -p ${BUILD_REPORTS_PATH} && cp -r build/reports/tests/test ${BUILD_REPORTS_PATH}
       env:
         BUILD_REPORTS_PATH: ~/buildReports/${{runner.os}}/java${{matrix.java}}/${{matrix.RUNTIME}}-${{matrix.RUNTIME_VERSION}}/
     - uses: actions/upload-artifact@v2
-      if: ${{ runner.os == 'Linux' && failure() }}
+      if: ${{ failure() }}
       with:
         name: buildReportsArtifact
         path: ~/buildReports
         retention-days: 3
-    - name: Copy build/report/tests/test for upload (Windows)
-      if: ${{ runner.os == 'Windows' && failure() }}
+
+# WINDOWS BUILDS
+  build-windows:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # test against latest update of each major Java version, as well as specific updates of LTS versions:
+        RUNTIME: [ol, wlp]
+        RUNTIME_VERSION: [21.0.0.3, 20.0.0.12]
+        java: [11, 8]
+        exclude:
+        - java: 8
+          RUNTIME_VERSION: 20.0.0.12
+        - java: 8
+          RUNTIME: wlp
+    name: ${{ matrix.RUNTIME }} ${{ matrix.RUNTIME_VERSION }}, Java ${{ matrix.java }}, Windows
+    steps:
+# Checkout repos
+    - name: Checkout ci.gradle
+      uses: actions/checkout@v2
+    - name: Setup Java ${{ matrix.java }}
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: ${{ matrix.java }}
+        cache: 'gradle'
+    # Moving and cloning to C: drive for Windows for more disk space
+    - name: Clone ci.ant, ci.common, ci.gradle repos to C drive
+      run: |
+        cp -r D:/a/ci.gradle/ci.gradle C:/ci.gradle
+        git clone https://github.com/OpenLiberty/ci.common.git C:/ci.common
+        git clone https://github.com/OpenLiberty/ci.ant.git C:/ci.ant
+# Cache mvn/gradle packages
+    - name: Cache Maven packages
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Cache Gradle packages
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+# Install dependencies
+    - name: Install ci.ant and ci.common
+      run: |
+        cd C:/ci.ant
+        mvn clean install
+        cd C:/ci.common
+        mvn clean install
+# Run tests
+    - name: Run tests with Gradle on Windows
+      if: ${{ runner.os == 'Windows' }}
```

Comment:
Is this check needed? Think we are in a Windows only section here.

Reply:
Ah, I missed this line, I removed it.